src/sds.c

```diff
@@ -370,17 +339,26 @@ sds sdsResize(sds s, size_t size) {
      * to do the copy only if really needed. Otherwise if the change is
      * huge, we manually reallocate the string to use the different header
      * type. */
-    if (oldtype==type || (type < oldtype && type > SDS_TYPE_8)) {
-        newsh = s_realloc(sh, oldhdrlen+size+1);
-        if (newsh == NULL) return NULL;
-        s = (char*)newsh+oldhdrlen;
-    } else {
-        newsh = s_malloc(hdrlen+size+1);
-        if (newsh == NULL) return NULL;
-        memcpy((char*)newsh+hdrlen, s, len);
-        s_free(sh);
-        s = (char*)newsh+hdrlen;
-        s[-1] = type;
+    int is_realloc_op = (oldtype==type || (type < oldtype && type > SDS_TYPE_8));
+    size_t newlen = is_realloc_op ? oldhdrlen+size+1 : hdrlen+size+1;
+    int noop = 0;
+    #if defined(USE_JEMALLOC)
```

Comment:
please add a comment explaining what it does and why we do it.

Reply:
After this change, we call `sdsRemoveFreeSpace` from `tryObjectEncoding` only for >32KB strings. 
Additionally, it is called for the querybuf, which is not immutable.
Are you referring to the function being used elsewhere, such as in `RM_TrimStringAllocation`?
If so, should I revert this change or wrap the function with an argument indicating whether the string is immutable?